src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -40,414 +39,454 @@
+		}
+
+		// Check if the form identifier with that version exists
+		ClientFormMetadata clientFormMetadata = clientFormService
+				.getClientFormMetadataByIdentifierAndVersion(formIdentifier, formVersion, isJsonValidator);
+		ClientFormService.CompleteClientForm completeClientForm = null;
+
+		long formId;
+
+		if (clientFormMetadata == null) {
+			// Get an older form version
+			clientFormMetadata = getMostRecentVersion(formIdentifier, isJsonValidator);
+			if (clientFormMetadata == null) {
+				return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+			} else {
+				formId = clientFormMetadata.getId();
+
+				if (strictSearch) {
+					if (clientFormMetadata.getVersion().equals(currentFormVersion)) {
+						return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+					} else {
+						return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+					}
+				}
+			}
+		} else {
+			formId = clientFormMetadata.getId();
+		}
+
+		ClientForm clientForm = clientFormService.getClientFormById(formId);
+		if (clientForm == null) {
+			return new ResponseEntity<>((String) null, HttpStatus.NOT_FOUND);
+		}
+
+		completeClientForm = new ClientFormService.CompleteClientForm(clientForm, clientFormMetadata);
+
+		return new ResponseEntity<>(objectMapper.writeValueAsString(completeClientForm), HttpStatus.OK);
+	}
+
+	private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier, boolean isJsonValidator) {
+		List<IdVersionTuple> availableFormVersions = clientFormService
+				.getAvailableClientFormMetadataVersionByIdentifier(formIdentifier, isJsonValidator);
+		DefaultArtifactVersion highestVersion = null;
+		IdVersionTuple highestIdVersionTuple = null;
+
+		for (IdVersionTuple availableFormVersion : availableFormVersions) {
+			DefaultArtifactVersion semanticFormVersion = new DefaultArtifactVersion(availableFormVersion.getVersion());
+
+			if (highestVersion == null || semanticFormVersion.compareTo(highestVersion) > 0) {
+				highestVersion = semanticFormVersion;
+				highestIdVersionTuple = availableFormVersion;
+			}
+		}
+
+		if (highestVersion == null) {
+			return null;
+		} else {
+			long formId = highestIdVersionTuple.getId();
+			return clientFormService.getClientFormMetadataById(formId);
+		}
+	}
+
+	@RequestMapping(method = RequestMethod.GET, path = "release-related-files")
+	private ResponseEntity<String> getAllFilesRelatedToRelease(@RequestParam(value = "identifier") String releaseIdentifier)
+			throws JsonProcessingException {
+		if (StringUtils.isBlank(releaseIdentifier)) {
+			return new ResponseEntity<>("Request parameter cannot be empty", HttpStatus.BAD_REQUEST);
+		}
+
+		List<ClientFormMetadata> clientFormMetadataList = new ArrayList<>();
+
+		Manifest manifest = manifestService.getManifest(releaseIdentifier);
+		if (manifest != null && StringUtils.isNotBlank(manifest.getJson())) {
+			JSONObject json = new JSONObject(manifest.getJson());
+			if (json.has(FORM_IDENTIFIERS)) {
```

Comment:
I see a lot of changes in these files & I'm having a lot of difficulty understanding what functionally changed & where this change was made in the code